api_app/script_analyzers/observable_analyzers/zoomeye.py

```diff
@@ -0,0 +1,93 @@
+import requests
+
+from api_app.exceptions import AnalyzerRunException, AnalyzerConfigurationException
+from api_app.script_analyzers import classes
+from intel_owl import secrets
+
+
+class ZoomEye(classes.ObservableAnalyzer):
+    base_url: str = "https://api.zoomeye.org/"
+
+    def set_config(self, additional_config_params):
+        self.api_key_name = additional_config_params.get("api_key_name", "ZOOMEYE_KEY")
+        self.search_type = additional_config_params.get("search_type", "NULL")
+
+        # Setting query filters
+        self.query = {}
+        host_search_filters = [
+            "app",
+            "ver",
+            "device",
+            "os",
+            "service",
+            "cidr",
+            "port",
+            "city",
+            "country",
+            "asn",
+        ]
+        web_search_filters = ["header", "keywords", "desc", "title", "site"]
+        if "query" in additional_config_params:
+            for filter in host_search_filters:
+                self.query[filter] = additional_config_params["query"].get(
+                    filter, "NULL"
+                )
+            for filter in web_search_filters:
+                self.query[filter] = additional_config_params["query"].get(
+                    filter, "NULL"
+                )
+
+        if self.observable_classification == "ip":
+            self.query["ip"] = self.observable_name
+        elif self.observable_classification == "domain" and self.search_type == "host":
```

Comment:
If someone analyze a domain, I would force the search type to "host". Otherwise you would miss a case in this conditional clause